parts/linux/cloud-init/artifacts/cse_cmd.sh

```diff
@@ -158,4 +158,6 @@ IS_KATA="{{IsKata}}"
 ARTIFACT_STREAMING_ENABLED="{{IsArtifactStreamingEnabled}}"
 SYSCTL_CONTENT="{{GetSysctlContent}}"
 PRIVATE_EGRESS_PROXY_ADDRESS="{{GetPrivateEgressProxyAddress}}"
+ENABLE_IMDS_RESTRICTION="{{EnableIMDSRestriction}}"
```

Comment:
@Bryce-Soghigian for changes to cse_cmd.sh, although only additive
@lilypan26 @Devinwong for bootstrap contract

Reply:
Do we have a way of producing an image (off PR?) that one can directly test with?

Reply:
(Also please tag @Azure/karpenter rather than @Bryce-Soghigian directly. Thanks!)